### PR TITLE
Add custom product catalog attribute type

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://issues.ibexa.co/browse/IBX-2326
| Versions      | 4.1, master

### CODE REVIEW

**Review by adamwojs — COMMENTED** [review]
It's good starting point however we need to cover the following extension points in this page:
* `\Ibexa\Contracts\ProductCatalog\Local\Attribute\OptionsFormMapperInterface`
* `\Ibexa\Contracts\ProductCatalog\Local\Attribute\OptionsValidatorInterface`
* `\Ibexa\Contracts\ProductCatalog\Local\Attribute\ValueValidatorInterface`

File: code_samples/catalog/custom_attribute_type/config/custom_services.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+            -   name: ibexa.product_catalog.attribute_type
+                alias: date
+    App\Attribute\Date\Form\DateValueFormMapper:
+        tags:
+            -   name: ibexa.product_catalog.attribute.form_mapper.value
+                type: date
+    App\Attribute\Date\DateValueFormatter:
+        tags:
+            -   name: ibexa.product_catalog.attribute.formatter.value
+                type: date
```
Comment (adamwojs) [inline code_samples/catalog/custom_attribute_type/config/custom_services.yaml:16]:
```suggestion
    app.product_catalog.attribute_type.date:
        class: Ibexa\ProductCatalog\Local\Repository\Attribute\AttributeType
        arguments:
            $identifier: 'date'
        tags:
            -   name: ibexa.product_catalog.attribute_type
                alias: date
    
    App\Attribute\Date\Form\DateValueFormMapper:
        tags:
            -   name: ibexa.product_catalog.attribute.form_mapper.value
                type: date
    
    
    App\Attribute\Date\DateValueFormatter:
        tags:
            -   name: ibexa.product_catalog.attribute.formatter.value
                type: date
```

File: code_samples/catalog/custom_attribute_type/src/Attribute/Date/Form/DateValueFormMapper.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Validator\Constraints as Assert;
+
+final class DateValueFormMapper implements ValueFormMapperInterface
+{
+    public function createValueForm(
+        string $name,
+        FormBuilderInterface $builder,
+        AttributeDefinitionAssignmentInterface $assignment,
+        array $context = []
```
Comment (adamwojs) [inline code_samples/catalog/custom_attribute_type/src/Attribute/Date/Form/DateValueFormMapper.php:19]: IMHO $context requires a little bit explanation here

**Review by adamwojs — CHANGES_REQUESTED** [review]
Article is still missing explenation/example of `\Ibexa\Contracts\ProductCatalog\Local\Attribute\OptionsValidatorInterface

**Review by adamwojs — APPROVED** [review]

File: docs/guide/catalog/create_custom_attribute_type.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+First, create `PercentAttributeOptionsType` that defines two options, `min` and `max`.
+Both those options need to be of `PercentType`.
+
+``` php hl_lines="16 22"
+[[= include_file('code_samples/catalog/custom_attribute_type/src/Attribute/Percent/PercentAttributeOptionsType.php') =]]
+```
+
+### Options form mapper
+
+Next, create a `PercentOptionsFormMapper` that maps the information that the user input in the form into attribute definition.
```
Comment (juskora) [inline docs/guide/catalog/create_custom_attribute_type.md:71]:
```suggestion
Next, create a `PercentOptionsFormMapper` that maps the information that the user inputs in the form into attribute definition.
```

**Review by juskora — APPROVED** [review]

**Pull request merged**